### PR TITLE
[HACDOCS-1016] Added ArgoCD plugin content

### PR DESCRIPTION
**Pull request by gtrivedi88**
Purpose: To resolve [HACDOCS-1016](https://issues.redhat.com/browse/HACDOCS-1016)

Version for cherrypicking: NA

Content for preview:  [Output]()

SME review: @karthikjeeyar

### CODE REVIEW

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 1]
```diff
@@ -1,322 +1,268 @@
...
+        pluginConfig:
+          argocd:
+            username: "${ARGOCD_USERNAME}"
+            password: "${ARGOCD_PASSWORD}"
+            appLocatorMethods:
+              - type: 'config'
+                instances:
+                  - name: argoInstance1
+                    url: "${ARGOCD_INSTANCE1_URL}"
+                    token: "${ARGOCD_AUTH_TOKEN}"
```
Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:178]: Safe to remove this, it is not required.
```suggestion
```

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 2]
```diff
@@ -1,322 +1,268 @@
...
+                        allOf:
+                          - isArgocdAvailable
+                  - mountPoint: entity.page.cd/cards
+                    importName: ArgocdDeploymentLifecycle
+                    config:
+                      layout:
+                        gridColumn: '1 / -1'
+                      if:
+                        allOf:
+                          - isArgocdConfigured
```
Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:203]: we can remove this.
```suggestion
```

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 3]
```diff
@@ -1,322 +1,268 @@
...
 ----
 
-Also, you can use Helm to install the ArgoCD plugin and be insecure by default:
+=== Loading the Plugin in Backstage Showcase Application
```
Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:206]: Can we add this link to the backstage-showcase repo somewhere here https://github.com/janus-idp/backstage-showcase ?

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 4]
```diff
@@ -1,322 +1,268 @@
-= ArgoCD plugin for {product}
+= Installation and Configuration of ArgoCD
 
-The ArgoCD plugin represents the current status of an application in your Backstage Catalog.
+You can use the ArgoCD plugin to visualize Continuous Delivery (CD) pipeline runs in OpenShift GitOps. This plugin provides a visual overview of the application’s status, deployment details, author of the pipeline run, commit message, and the container image promoted to production.
```
Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:3]:
```suggestion
You can use the ArgoCD plugin to visualize the Continuous Delivery (CD) workflows in OpenShift GitOps. This plugin provides a visual overview of the application’s status, deployment details, commit message, author of the commit, container image promoted to environment and deployment history.
```

**Review by karthikjeeyar — COMMENTED** [review]
Can you please add some screenshots of the plugin, so the users know what will be available to them post configuration ?

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc (hunk 3 above)
Comment (karthikjeeyar) [reply artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:206]: I was thinking more about this section, this section is about running the backstage-showcase repo locally by some developers, I dont think we need this section here in RHDH at all, we can remove this section.

Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:206]:
```suggestion
```

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 5]
```diff
@@ -1,322 +1,191 @@
...
-
-The *ArgoCD history* page contains history of application instances along with other information, such as deploy details, author name, message that author added, and revision ID.
---
-
-
+image::rhdh-plugins-reference/argocd.png[CD tab ArgoCD]
 
+[NOTE]
+====
+For additional deployment details, select an appropriate card or select the link icon (image:rhdh-plugins-reference/link.svg[width=12px, height=12px, align=center]) to open the deployment details in *argo*.
```
Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:182]: Can we break it into two points, bcoz they are different operations.
```suggestion
1. For additional deployment details, click on a card to open the sidebar.
2. Click on the link icon (image:rhdh-plugins-reference/link.svg[width=12px, height=12px, align=center]) to open the deployment details in *Argo CD*.
```

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 6]
```diff
@@ -1,322 +1,191 @@
...
+==== 
 
+. (Optional) Alternatively, select *Overview* tab, and then from Deployment summary section, select an appropriate application to open the deployment details in *argo*.
```
Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:185]:
```suggestion
 *Overview* tab contains the Deployment summary, select an appropriate application to open it in *Argo CD*.
```

Can we also add the screenshot of the Overview tab here ?


image

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 7]
```diff
@@ -1,322 +1,191 @@
-= ArgoCD plugin for {product}
+= Installation and Configuration of ArgoCD
```
Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:1]: Can we use the name to Argo CD with a space throughout this doc?  Even in backstage-plugins repo, there is a PR to update the naming with a space. 

```suggestion
= Installation and Configuration of Argo CD plugin
```

**Review by karthikjeeyar — APPROVED** [review]
Thanks @gtrivedi88 for all the updates.

/lgtm

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 8]
```diff
@@ -1,322 +1,114 @@
...
-        - name: argoInstance1
-          url: https://argoInstance1.com
-          token: ${ARGOCD_AUTH_TOKEN} # Token to use to instance 1
-----
---
-
-=== Permissions in ArgoCD plugin
-
-Setting permissions for ArgoCD user account not only reduces the scope but also reduces the functionality of the backend. If you scope the permissions for read-only, then actions including creating, deleting, and resyncing the application will not be available. 
+* Select an appropriate card. The system displays the deployment details (for example, commit history, author name, and deployment history) in a sidebar.
```
Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:97]:
```suggestion
* Select an appropriate card. The system displays the deployment details (for example, commit message, author name, and deployment history) in a sidebar.
```

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 9]
```diff
@@ -1,322 +1,114 @@
...
-
-Creating a `.npmrc` file ensures that all the packages are scoped under `@redhat` and are fetched from link:https://npm.registry.redhat.com/[Red Hat's NPM registry], while the rest dependencies remain sourced from other link:registry.npmjs.org[registry]. 
+=== Enabling Argo CD plugin
 
 .Prerequisites
 
-* Your Backstage application is installed and deployed.
-
-.Procedure
+* Add Argo CD instance information to your `app.config.yaml` file:
```
Comment (karthikjeeyar) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:11]:
```suggestion
* Add Argo CD instance information to your `app.config.yaml` configmap:
```

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 10]
```diff
@@ -1,322 +1,114 @@
...
-
-Creating a `.npmrc` file ensures that all the packages are scoped under `@redhat` and are fetched from link:https://npm.registry.redhat.com/[Red Hat's NPM registry], while the rest dependencies remain sourced from other link:registry.npmjs.org[registry]. 
+=== Enabling Argo CD plugin
 
 .Prerequisites
 
-* Your Backstage application is installed and deployed.
-
-.Procedure
+* AAdd Argo CD instance information to your `app.config.yaml` configmap:
```
Comment (subhashkhileri) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:11]: There is a typo `AAdd`

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 11]
```diff
@@ -1,322 +1,114 @@
...
-[source]
-----
-// packages/app/src/plugins.ts
-export { argocdPlugin } from '@redhat/backstage-plugin-argo-cd';
+[source,yaml]
 ----
---
+ annotations:
+   ...
+    # The Argo CD instance name used in `app.config.yaml`.
```
Comment (subhashkhileri) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:49]: Update config file name `app.config.yaml` to `app-config.yaml`

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 12]
```diff
@@ -1,322 +1,114 @@
...
-`argocd/app-name: <app-name>`
-
-To select multiple ArgoCD applications for a component, use labels as follows:
-
-`argocd/app-selector: <app-selector>`
-
 [NOTE]
 ====
-You can either use annotations or labels for a component.
+If you do not set this annotation, the Argo CD plugin defaults to the first Argo CD instance configured in `app.config.yaml`.
```
Comment (subhashkhileri) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:57]: Update config file name app.config.yaml to app-config.yaml

**Review by Gerry-Forde — APPROVED** [review]

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 13]
```diff
@@ -1,322 +1,114 @@
...
-
 [NOTE]
 ====
-You can either use annotations or labels for a component.
+If you do not set this annotation, the Argo CD plugin defaults to the first Argo CD instance configured in `app-config.yaml`.
 ====
---
 
-. Send a GET HTTP request to ArgoCD’s `/session` endpoint using the username and password to acquire your authentication key.
+.Procedures
```
Comment (linfraze) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:60]: Procedure

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 14]
```diff
@@ -1,322 +1,114 @@
...
-      instances:
-        - name: argoInstance1
-          url: https://argoInstance1.com
-          token: ${ARGOCD_AUTH_TOKEN} # optional
-        - name: argoInstance2
-          url: https://argoInstance2.com
-          # dedicated username/password for this instance
-          username: ${ARGOCD_USERNAME_INSTANCE_2} # optional
-          password: ${ARGOCD_PASSWORD_INSTANCE_2} # optional
+. On the {product-very-short} platform, navigate to the *Catalog* and select an appropriate component.
```
Comment (linfraze) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:87]: Click the **Catalog** tab and select the component that you want to use.

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 15]
```diff
@@ -1,322 +1,114 @@
...
 
-----
-
-A loop is created between deleting ArgoCD project and application to check the deletion of application occurring before the deletion of the project. If waitCycles is set to 25, then the loop can last for 75 seconds before erroring out.
-
-With the previous configuration, the ArgoCD plugin fetches the ArgoCD instances an application is deployed to and use the ArgoCD backend plugin (`backstage-plugin-argo-cd-backend`) to reach out to each ArgoCD instance based on the following mapping:
+. Select *CD* tab to view insights into deployments managed by Argo CD.
```
Comment (linfraze) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:89]: Click the **CD** tab

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 16]
```diff
@@ -1,322 +1,114 @@
...
-
-The ArgoCD overview card displays application information including *NAME*, *INSTANCE*, *SYNC STATUS*, *HEALTH STATUS*, and *LAST SYNCED*.
---
-
-. To view the detailed application information, select an application from the ArgoCD overview card.
-+
---
-A pop-up containing detailed information about the application appears, along with a link to ArgoCD dashboard.
++ 
+* Select the link icon (image:rhdh-plugins-reference/link.svg[width=13px, height=13px, align=center]) to open the deployment details in Argo CD.
```
Comment (linfraze) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:103]: Remove For deployment details

Format this as step 3: Optional: Click the [object] to view its deployment details, for example, commit message, author, name, and deployment history.
Format this is as a sub-step to step 3: Optional: Click the link icon to open the deployment details in Argo CD.

Use _click_ and _select_ in accordance with IBM stlye guideance throughout the doc

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 17]
```diff
@@ -1,322 +1,114 @@
...
 
-image::argo-cd-plugin-overview-card-details.png[ArgoCD overview card]
---
+. Select the *Overview* tab and navigate to the Deployment summary section to review the summary of your application's deployment across namespaces. Additionally, select an appropriate Argo CD app to open the deployment details in Argo CD, or select a commit from the Revision column to review the changes in GitLab or GitHub.
```
Comment (linfraze) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:105]: Click the **Overview** tab

If Deployment summary is a heading on the UI, then bold it. Same for Revision.

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc [hunk 18]
```diff
@@ -1,322 +1,114 @@
...
-
-
-
+image::rhdh-plugins-reference/deployment_summary.png[Deployment summary]
 
 
+[role="_additional-resources"]
+.Additional resources
 
+* For more information on dynamic plugins, refer link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.2/html/configuring_plugins_in_red_hat_developer_hub/rhdh-installing-dynamic-plugins[Dynamic plugin installation].
```
Comment (linfraze) [inline artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:114]: Change _refer_ to _see_ to align with the rest of the RHDH docs. It also looks like "to" is missing after "refer" for _refer to_

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc (hunk 15 above)
Comment (gtrivedi88) [reply artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:89]: @linfraze Thank you for your suggestion. I decided to use "Select" based on this:
image

Let me know if you disagree or have some other references.
Thanks again.

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc (hunk 17 above)
Comment (gtrivedi88) [reply artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:105]: Thank you!
Not sure if I understand this. Deployment summary is a section name that the system displays on the Overview tab. Do we need to bold section names? ..same for column names, like Revision.

Please suggest.

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc (hunk 16 above)
Comment (gtrivedi88) [reply artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:103]: > Remove For deployment details
Updated

> Format this as step 3: Optional: Click the [object] to view its deployment details, for example, commit message, author, name, and deployment history.
> Format this is as a sub-step to step 3: Optional: Click the link icon to open the deployment details in Argo CD.

Updated

> Use click and select in accordance with IBM stlye guideance throughout the doc
Raised question above

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc (hunk 14 above)
Comment (gtrivedi88) [reply artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:87]: > Click the Catalog tab and select the component that you want to use.
Updated but raised a question below. Click vs Select. Kept select as for now

**Comment by gtrivedi88** [comment]
@linfraze Thank you for the comments. Raised a new PR addressing your comments: https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/413
Also, requested some suggestions on your comments

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc (hunk 15 above)
Comment (gtrivedi88) [reply artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:89]: Based on this discussion - https://redhat-internal.slack.com/archives/C04P5S34GP2/p1722440235254279
...going with Select.

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc (hunk 17 above)
Comment (gtrivedi88) [reply artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:105]: Based on this discussion - https://redhat-internal.slack.com/archives/C04P5S34GP2/p1722440235254279
...going with Select.

If you feel this is not right, we can discuss more :)

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc (hunk 15 above)
Comment (linfraze) [reply artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:89]: @gtrivedi88 that's fine. I appreciate you starting the discussion in the CCS channel. We all learn a lot that way.

File: artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc (hunk 17 above)
Comment (linfraze) [reply artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:105]: @gtrivedi88 I was basing the recommendation to bold the UI elements on [this guidance](https://redhat-documentation.github.io/supplementary-style-guide/#user-interface-elements), but use your judgement

**Pull request merged**